client/src/components/Products/Products.tsx

```diff
@@ -1,31 +1,12 @@
 import style from './styles.module.scss';
 import { useEffect, useState } from 'react';
 import axios from 'axios';
+import { ProductData } from '@/types/interfaces';
+import { ProductsType } from '@/types/interfaces';
 import { FC } from 'react';
 
-interface ProductData {
-  id: number;
-  attributes: {
-    title: string;
-    disc: string;
-    price: number;
-    img: {
-      data: {
-        attributes: {
-          url: string;
-        };
-      };
-    };
-  };
-}
-
-interface Products {
-  numberOfProducts: number;
-}
-
-export const Products: FC<Products> = ({ numberOfProducts }) => {
+export const Products: FC<ProductsType> = ({ numberOfProducts }) => {
```

Comment:
interface name for component props shuold have suffix props

ProductProps or IProductProps (you chose not to add I prefix it's possible, but use suffix props)

same as in other components leave interface for props above the component function